src/components/Grid/useDataGridSource.tsx

```
@@ -253,23 +275,29 @@ export const useDataGridSource = <
+        ...persistedFilterModel,
+      };
+      return {
+        ...defaultInitialInput,
+        ...initialInputRef.current,
+        count: initialInputRef.current?.count ?? defaultInitialInput.count,
+        ...(view.apiSortModel?.[0] && {
+          sort: view.apiSortModel[0].field,
+          order: upperCase(view.apiSortModel[0].sort!),
+        }),
+        // eslint-disable-next-line no-extra-boolean-cast
+        filter: Boolean(apiRef.current.instanceId)
```

Biome

[error] Avoid redundant `Boolean` call

It is not necessary to use `Boolean` call when a value will already be coerced to a boolean.
Unsafe fix: Remove redundant `Boolean` call


(lint/complexity/noExtraBooleanCast)